api/v1beta1/azureclustertemplate_default_test.go

```diff
@@ -354,6 +360,9 @@ func TestSubnetsTemplateDefaults(t *testing.T) {
 											Role:       SubnetNode,
 											CIDRBlocks: []string{DefaultNodeSubnetCIDR},
 										},
+										NatGateway: NatGatewayClassSpec{
+											Name: DefaultNATGateway,
+										},
```

Comment:
Can you add a new test case for when both IPv6 and NatGateway is enabled? This way, you can cover the case of setting the NatGateway defaults with a Subnet

Reply:
Thanks @willie-yao for bringing this up! NatGateway is not supported by IPv6 cluster, so I am worry if this is an valid case or not where NatGateway and IPv6 both are enabled.
BTW, I think below case: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/81f88115494b6c390621641e6074a760877b4074/api/v1beta1/azureclustertemplate_default_test.go#L381-L436 has covered the negative test for IPv6. This is a cluster enables IPv6, so NatGateway will not be set for AzureClusterTemplate.